homeassistant/components/netgear/device_tracker.py

```diff
@@ -39,132 +40,90 @@
-                    or dev.name in self.excluded_devices
-                )
-            )
-            if tracked:
-                devices.append(dev.mac)
-                if (
-                    self.tracked_accesspoints
-                    and dev.conn_ap_mac in self.tracked_accesspoints
-                ):
-                    devices.append(f"{dev.mac}_{dev.conn_ap_mac}")
-
-        return devices
-
-    def get_device_name(self, device):
-        """Return the name of the given device or the MAC if we don't know."""
-        parts = device.split("_")
-        mac = parts[0]
-        ap_mac = None
-        if len(parts) > 1:
-            ap_mac = parts[1]
-
-        name = None
-        for dev in self.last_results:
-            if dev.mac == mac:
-                name = dev.name
-                break
-
-        if not name or name == "--":
-            name = mac
-
-        if ap_mac:
-            ap_name = "Router"
-            for dev in self.last_results:
-                if dev.mac == ap_mac:
-                    ap_name = dev.name
-                    break
-
-            return f"{name} on {ap_name}"
-
-        return name
-
-    def _update_info(self):
-        """Retrieve latest information from the Netgear router.
-
-        Returns boolean if scanning successful.
-        """
-        _LOGGER.debug("Scanning")
-
-        results = self.get_attached_devices()
-
-        if _LOGGER.isEnabledFor(logging.DEBUG):
-            _LOGGER.debug("Scan result: \n%s", pformat(results))
-
-        if results is None:
-            _LOGGER.warning("Error scanning devices")
-
-        self.last_results = results or []
-
-    def get_attached_devices(self):
-        """List attached devices with pynetgear.
-
-        The v2 method takes more time and is more heavy on the router
-        so we only use it if we need connected AP info.
-        """
-        if self.tracked_accesspoints:
-            return self._api.get_attached_devices_2()
-
-        return self._api.get_attached_devices()
+async def async_get_scanner(hass, config):
+    """Import Netgear configuration from YAML."""
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data=config,
+        )
+    )
+
+    _LOGGER.warning(
+        "Your Netgear configuration has been imported into the UI, "
+        "please remove it from configuration.yaml. "
+        "Loading Netgear via platform setup is now deprecated"
+    )
+
+    return None
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up device tracker for Netgear component."""
+
+    def generate_classes(router: NetgearRouter, device):
+        classes = []
+        classes.append(NetgearScannerEntity(router, device))
+        return classes
+
+    await async_setup_netgear_entry(hass, entry, async_add_entities, generate_classes)
```

Comment:
```suggestion
    await async_setup_netgear_entry(hass, entry, async_add_entities, [NetgearScannerEntity(router, device)])
```

Reply:
This does not work since the `generate_classes` is a function that will be called in the `async_setup_netgear_entry` class, a list cannot be called, so this will result in an error.